src/category_theory/sites/compatible_sheafification.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2021 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import category_theory.sites.compatible_plus
+
+/-!
+
+In this file, we prove that sheafification is compatible with functors which
+preserve the correct limits and colimits.
+
+-/
+
+namespace category_theory.grothendieck_topology
+
+open category_theory
+open category_theory.limits
+open opposite
+
+universes w₁ w₂ v u
+variables {C : Type u} [category.{v} C] (J : grothendieck_topology C)
+variables {D : Type w₁} [category.{max v u} D]
+variables {E : Type w₂} [category.{max v u} E]
+variables (F : D ⥤ E)
+
+noncomputable theory
+
+variables [∀ (α β : Type (max v u)) (fst snd : β → α),
+  has_limits_of_shape (walking_multicospan fst snd) D]
+variables [∀ (α β : Type (max v u)) (fst snd : β → α),
+  has_limits_of_shape (walking_multicospan fst snd) E]
+variables [∀ (X : C), has_colimits_of_shape (J.cover X)ᵒᵖ D]
+variables [∀ (X : C), has_colimits_of_shape (J.cover X)ᵒᵖ E]
+variables [∀ (X : C), preserves_colimits_of_shape (J.cover X)ᵒᵖ F]
+variables [∀ (X : C) (W : J.cover X) (P : Cᵒᵖ ⥤ D), preserves_limit (W.index P).multicospan F]
+
+variables (P : Cᵒᵖ ⥤ D)
+
+/-- The isomorphism between the sheafification of `P` composed with `F` and
+the sheafification of `P ⋙ F`. -/
+def sheafify_comp_iso : J.sheafify P ⋙ F ≅ J.sheafify (P ⋙ F) :=
```

Comment:
Should all these defs come with simps?

Reply:
I'm not sure how useful these would be in practice. I think it would be much more useful to relate these isomorphism to `sheafify_lift`, but that requires additional assumptions to know that sheafification is a sheaf. What do you propose? I could add such lemmas in a separate section with these additional hypotheses, but I think simply adding `simps` here would generate lemmas that unfold things "too much".

Reply:
I added lemmas relating hom and inv of this construction to sheafify_lift and to to_sheafify, as well as some simp lemmas reducing the other isomorphisms in this file to these cases... take a look toward the bottom of the file.